pkg/controller/noderesource/node_resource_controller.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package noderesource
+
+import (
+	"context"
+	"github.com/go-logr/logr"
+	predictionapi "github.com/gocrane/api/prediction/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/version"
+	"k8s.io/client-go/scale"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strconv"
+)
+
+// NodeResourceReconciler reconciles a NodeResource object
+type NodeResourceReconciler struct {
+	client.Client
+	Scheme 		*runtime.Scheme
+	Log         logr.Logger
+	RestMapper  meta.RESTMapper
+	Recorder    record.EventRecorder
+	scaleClient scale.ScalesGetter
+	K8SVersion  *version.Version
+}
+
+//+kubebuilder:rbac:groups=node-resource.crane.io,resources=noderesources,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=node-resource.crane.io,resources=noderesources/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=node-resource.crane.io,resources=noderesources/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the NodeResource object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.10.0/pkg/reconcile
+func (r *NodeResourceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	r.Log.Info("node resource reconcile", "node-resource", req.NamespacedName, req.Name)
+
+	nodePre := &predictionapi.NodePrediction{}
+	// 1.取预测结果  nodePre.Status
+	err := r.Client.Get(ctx, req.NamespacedName, nodePre)
+	if err != nil {
+		r.Recorder.Event(nodePre, v1.EventTypeNormal, "FailedGetNodePrediction", err.Error())
```

Comment:
please also add event for key steps of successful path.

Reply:
ok